File: 2026-08/src/day_25/prms_120835/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> emergency) {
    int n = emergency.size();

    vector<int> idx(n);
    iota(idx.begin(), idx.end(), 0);
    sort(idx.begin(), idx.end(), [&](int a, int b) {
        return emergency[a] > emergency[b];
    });

    vector<int> ans(n);
    for (int i = 0; i < n; i++) {
        ans[idx[i]] = i + 1;
    }

    return ans;
}

File: 2026-08/src/day_25/prms_120835/Solution.java
package day_25.prms_120835;

import java.util.*;

class Solution {
    public int[] solution(int[] emergency) {
        int n = emergency.length;

        Integer[] idx = new Integer[n];
        for (int i = 0; i < n; i++) {
            idx[i] = i;
        }
        Arrays.sort(idx, (o1, o2) -> Integer.compare(emergency[o2], emergency[o1]));

        int[] ans = new int[n];
        for (int i = 0; i < n; i++) {
            ans[idx[i]] = i + 1;
        }

        return ans;
    }
}
